operations.go

```diff
@@ -0,0 +1,67 @@
+package webrtc
+
+import (
+	"sync"
+)
+
+// Operation is a function
+type operation func()
+
+// Operations is a task executor.
+type operations struct {
+	ops     []operation
+	mu      sync.Mutex
+	startMu sync.Mutex
+	done    chan struct{}
+}
+
+// Enqueue adds a new action to be executed. If there are no actions scheduled,
+// the execution will start immediately in a new goroutine.
+func (o *operations) Enqueue(op operation) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	o.ops = append(o.ops, op)
+	if len(o.ops) == 1 {
```

Comment:
I don't think we have a race, but just double checking.

* start drains the last operation.
* Enqueue is called, `len(o.ops) == 1`, we just added a new operation
* `start` will start pulling the new operation, meaning we don't exit.

This will cause us to have two `o.done` in flight. We should probably pop `o.ops` and return under the same lock. Otherwise someone could sneak in another operation.

Reply:
I hadn't thought about adding an `isLast` / `hasMore` flag before. I just updated the code to be:

```golang
func (o *operations) pop() (fn func(), isLast bool) {
	o.mu.Lock()
	defer o.mu.Unlock()
	fn = o.ops[0]
	o.ops = o.ops[1:]
	return fn, len(o.ops) == 0
}

func (o *operations) start() {
	var fn func()
	isLast := false
	for !isLast {
		fn, isLast = o.pop()
		fn()
	}
}
```

I think this should take care of it?